File: tests/fixtures/hello-static.cc
#include <cstdint>
#include <cstdio>
#include <cstdlib>
#include <cstring>

static std::uint32_t checksum(const char *text) {
    std::uint32_t value = 2166136261u;
    while (*text != '\0') {
        value ^= static_cast<std::uint8_t>(*text);
        value *= 16777619u;
        ++text;
    }
    return value;
}

int main(int argc, char **argv) {
    const char *environment = std::getenv("PACKFORGE_SMOKE");
    if (argc != 2 || environment == nullptr) {
        std::fputs("expected one argument and PACKFORGE_SMOKE\n", stderr);
        return 2;
    }

    std::printf("packforge-smoke argc=%d arg=%s env=%s checksum=%u\n", argc,
                argv[1], environment,
                checksum(argv[1]) ^ checksum(environment));
    return std::strcmp(argv[1], "round-trip") == 0 ? 0 : 3;
}

File: tests/fixtures/hello-static.go
package main

import (
	"fmt"
	"os"
)

func checksum(text string) uint32 {
	value := uint32(2166136261)
	for index := 0; index < len(text); index++ {
		value ^= uint32(text[index])
		value *= 16777619
	}
	return value
}

func main() {
	environment, present := os.LookupEnv("PACKFORGE_SMOKE")
	if len(os.Args) != 2 || !present {
		fmt.Fprintln(os.Stderr, "expected one argument and PACKFORGE_SMOKE")
		os.Exit(2)
	}

	argument := os.Args[1]
	fmt.Printf("packforge-smoke argc=%d arg=%s env=%s checksum=%d\n",
		len(os.Args), argument, environment, checksum(argument)^checksum(environment))
	if argument != "round-trip" {
		os.Exit(3)
	}
}
